keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/plugin/ResourceHandler.kt

```diff
@@ -29,10 +34,51 @@ interface ResourceHandler<T : Any> : KeelPlugin {
    */
   val supportedKind: Pair<ResourceKind, Class<T>>
 
+  val objectMapper: ObjectMapper
+  val idGenerator: ULID
+
+  /**
+   * Validates the resource spec, and generates a metadata header, and applies any defaults /
+   * normalization.
+   *
+   * Implementors should not generally need to override this method.
+   *
+   * @return a hydrated `Resource` with a name generated by convention, a `uid`, default values
+   * applied, etc.
+   */
+  fun validate(resource: SubmittedResource<*>): Resource<T> {
```

Comment:
The `ResourceHandler` interface can do the default stuff - constructing metadata & parsing the spec. Don't need to re-implement it in every handler.